packages/web_benchmarks/lib/src/recorder.dart

```diff
@@ -942,18 +965,25 @@ class Profile {
       final Timeseries timeseries = scoreData[key]!;
 
       if (timeseries.isReported) {
-        scoreKeys.add('$key.average');
+        scoreKeys.add('$key.${BenchmarkMetricComputation.average.name}');
         // Report `outlierRatio` rather than `outlierAverage`, because
         // the absolute value of outliers is less interesting than the
         // ratio.
-        scoreKeys.add('$key.outlierRatio');
+        scoreKeys.add('$key.${BenchmarkMetricComputation.outlierRatio.name}');
       }
 
       final TimeseriesStats stats = timeseries.computeStats();
-      json['$key.average'] = stats.average;
-      json['$key.outlierAverage'] = stats.outlierAverage;
-      json['$key.outlierRatio'] = stats.outlierRatio;
-      json['$key.noise'] = stats.noise;
+      json['$key.${BenchmarkMetricComputation.average.name}'] = stats.average;
+      json['$key.${BenchmarkMetricComputation.outlierAverage.name}'] =
+          stats.outlierAverage;
+      json['$key.${BenchmarkMetricComputation.outlierRatio.name}'] =
+          stats.outlierRatio;
+      json['$key.${BenchmarkMetricComputation.noise.name}'] = stats.noise;
+      for (final MapEntry<double, double> percentileEntry
```

Comment:
Should we instead iterate over `PercentileMetricComputation.values` and extract computed data using `stats.percentiles[metric.percentile]`? Then we don't need to reconstruct the name of the metric from its `double` value. In fact the metric name can be anything human-readable, like "95th percentile". One issue with `double.toString` is that JS behavior can be different from Wasm/VM behavior for certain values.

Sketch:

```dart
for (final metric in PercentileMetricComputation.values) {
  jaons[metric.name] = stats.percentiles[metric.percentile]!;
}
```